### ISSUE TITLE
Albert does not use SDPA's Flash Attention since attention mask is always created

### ISSUE THREAD

**Issue by gau-nernst** [huggingface/transformers#36217]
### System Info

NA

### Who can help?

@ArthurZucker 

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

https://github.com/huggingface/transformers/blob/dd16acb8a3e93b643aa374c9fb80749f5235c1a6/src/transformers/models/albert/modeling_albert.py#L772-L773

Since attention mask is always used by `AlbertModel`, SDPA will dispatch memory-efficient kernel instead of flash attention. The inner classes (`AlbertTransformer`, `AlbertLayerGroup`, ...) seem to handle attention_mask=None correctly, so it's only the outermost class is problematic.

### Expected behavior

Flash attention should be used when attention_mask=None

**Comment by gau-nernst** [comment]
Edit: After more careful investigation, my example didn't use Flash attention because I was running it in FP32. `_prepare_4d_attention_mask_for_sdpa()` actually returns None if attention mask is all ones -> it will use flash attention as expected. Though there is a small inefficiency in creating a ones mask just to check it and return None instead.

https://github.com/huggingface/transformers/blob/dd16acb8a3e93b643aa374c9fb80749f5235c1a6/src/transformers/modeling_attn_mask_utils.py#L448-L449
